backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interest/InterestComputationCommand.java

```diff
@@ -222,42 +223,42 @@ private ImmutableSet<FlatrateTermId> getModularContractIds(
 			@NonNull final InterestComputationRequest request)
 	{
 		return modularContractLogService.getModularContractIds(ModularContractLogQuery.builder()
-				.computingMethodType(request.getComputingMethodType())
-				.invoicingGroupId(request.getInvoicingGroupId())
-				.lockOwner(request.getLockOwner())
-				.processed(false)
-				.billable(true)
-				.build());
+																	   .computingMethodType(request.getComputingMethodType())
+																	   .invoicingGroupId(request.getInvoicingGroupId())
+																	   .lockOwner(request.getLockOwner())
+																	   .processed(false)
+																	   .billable(true)
+																	   .build());
 	}
 
-	private void saveAddedValueInterestRecords(@NonNull final InvoiceAllocations currentInvoiceAllocations)
+	private void saveAddedValueInterestRecords(@NonNull final InterimContractAllocations currentInterimContractAllocations)
 	{
-		currentInvoiceAllocations.getAllocatedShippingNotifications().forEach(interestRepository::create);
+		currentInterimContractAllocations.getAllocatedShippingNotifications().forEach(interestRepository::create);
 	}
 
 	private void saveSubtractValueInterestRecord(
 			@NonNull final InterestComputationRequest request,
 			@NonNull final Percent bonusInterestRate,
-			@NonNull final InvoiceAllocations.AllocationItem shippingNotification)
+			@NonNull final InterimContractAllocations.AllocationItem shippingNotification)
 	{
-		if (request.getBonusComputationDetails() == null)
+		if (request.getBonusComputationTimeInterval() == null)
 		{
 			final CreateModularLogInterestRequest createInterestRequest = CreateModularLogInterestRequest.builder()
 					.interestRunId(request.getInterestRunId())
 					.shippingNotificationLogId(shippingNotification.getShippingNotificationEntry().getId())
 					.allocatedAmt(shippingNotification.getOpenAmount())
-					.interestDays(0)
+					.interestDays((long)0)
 					.finalInterest(Money.zero(request.getInterestToDistribute().getCurrencyId()))
```

Comment:
I would take out this as ca constant and give a proper name, that could explain what's the 0 purpose